reformatted var function: add examples, add type hint for keepdims

egg-west opened the pull request:
#2012

egg-west commented:
The lint check complains about the length of one line (line 492 in ivy/ivy/functional/ivy/statistical.py. I added this line with the guidance of docstring; Should one of them be updated? (lint check or docstring guidance).

RitujaPawas commented:
> The lint check complains about the length of one line (line 492 in ivy/ivy/functional/ivy/statistical.py. I added this line with the guidance of docstring; Should one of them be updated? (lint check or docstring guidance).

You should fix the length of line 492 so that the length is always 88 characters or less.
and pull the latest changes for now. 

Thanks for flagging, will look into why lint errors in docstrings are not automatically fixed in files.

RitujaPawas commented on ivy/functional/ivy/statistical.py:
```diff
@@ -487,6 +487,81 @@ def var(
+    >>> y = ivy.var(x, axis=0)
+    >>> print(y)
+    ivy.array([6., 6., 6.])
+
```
Please add more functional examples according to [docstring-examples-guide](https://lets-unify.ai/ivy/deep_dive/12_docstring_examples.html#docstring-examples) here.

egg-west replied:
I missed some points. Updated.

RitujaPawas replied:
Add docstring and docstring examples to `var` in `ivy/array/statistical.py` as well as in `ivy/container/statistical.py`

RitujaPawas replied:
Make sure to request review once you have made the changes.

egg-west replied:
Updated, please help and review.

egg-west commented on ivy/functional/backends/jax/statistical.py:
```diff
@@ -109,7 +109,7 @@ def var(
     x: JaxArray,
     axis: Optional[Union[int, Tuple[int, ...]]] = None,
```
Good standard. Changed.

egg-west commented:
Hi, I need help pulling the latest version from upstream. Could you @RitujaPawas  guide me to do this? Thanks!

RitujaPawas commented:
> Hi, I need help pulling the latest version from upstream. Could you @RitujaPawas guide me to do this? Thanks!

Hey, did you follow through the [Contributing guide](https://lets-unify.ai/ivy/contributing/1_the_basics.html#managing-your-fork) ? Let me know if you still face difficulties!

egg-west commented:
> 

Thanks! I was trying to pull the commits and rebase on the latest version of upstream. Because this is necessary before we finish this pull request. 

Concretely, I tried 1) pull upstream to my local repository. 2) try to push the updated local repository to my fork but get rejected. 
Here are my commands and printings:

(base) linjx@ethans-macbook ivy % git pull upstream master
remote: Enumerating objects: 1601, done.
remote: Counting objects: 100% (853/853), done.
remote: Compressing objects: 100% (40/40), done.
remote: Total 1601 (delta 816), reused 846 (delta 813), pack-reused 748
Receiving objects: 100% (1601/1601), 444.03 KiB | 749.00 KiB/s, done.
Resolving deltas: 100% (1197/1197), completed with 197 local objects.
From https://github.com/unifyai/ivy
 * branch                master     -> FETCH_HEAD
 + b9826c337...eb5b9523b master     -> upstream/master  (forced update)
hint: You have divergent branches and need to specify how to reconcile them.
hint: You can do so by running one of the following commands sometime before
hint: your next pull:
hint: 
hint:   git config pull.rebase false  # merge
hint:   git config pull.rebase true   # rebase
hint:   git config pull.ff only       # fast-forward only
hint: 
hint: You can replace "git config" with "git config --global" to set a default
hint: preference for all repositories. You can also pass --rebase, --no-rebase,
hint: or --ff-only on the command line to override the configured default per
hint: invocation.
fatal: Need to specify how to reconcile divergent branches.

(base) linjx@ethans-macbook ivy % git config pull.rebase true

(base) linjx@ethans-macbook ivy % git pull upstream master   
From https://github.com/unifyai/ivy
 * branch                master     -> FETCH_HEAD
Successfully rebased and updated refs/heads/master.

(base) linjx@ethans-macbook ivy % git push
To https://github.com/egg-west/ivy.git
 ! [rejected]            master -> master (non-fast-forward)
error: failed to push some refs to 'https://github.com/egg-west/ivy.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
(base) linjx@ethans-macbook ivy % git pull
dropping f98c02e5a593d7e13a8ed6917f316fdf7d27d4da Fix pr_dash lint issues -- patch contents already upstream
Auto-merging ivy/functional/backends/numpy/creation.py
CONFLICT (content): Merge conflict in ivy/functional/backends/numpy/creation.py
error: could not apply b919f3998... Add 'out' to each function of creation in numpy backend
hint: Resolve all conflicts manually, mark them as resolved with
hint: "git add/rm <conflicted_files>", then run "git rebase --continue".
hint: You can instead skip this commit: run "git rebase --skip".
hint: To abort and get back to the state before "git rebase", run "git rebase --abort".
Could not apply b919f3998... Add 'out' to each function of creation in numpy backend
(base) linjx@ethans-macbook ivy % git pull
error: Pulling is not possible because you have unmerged files.
hint: Fix them up in the work tree, and then use 'git add/rm <file>'
hint: as appropriate to mark resolution and make a commit.
fatal: Exiting because of an unresolved conflict.


(base) linjx@ethans-macbook ivy % git status
interactive rebase in progress; onto b9826c337
Last commands done (56 commands done):
   pick 54148c40b Improving PR dashboard filtering
   pick b919f3998 Add 'out' to each function of creation in numpy backend
  (see more in file .git/rebase-merge/done)
Next commands to do (111 remaining commands):
   pick c6e8d4dd2 Add out arg to the creation submodule of jax, tf, torch backends
   pick fbab973fe Add whitespaces around equals; delete out arg of meshgrid
  (use "git rebase --edit-todo" to view and edit)
You are currently rebasing branch 'master' on 'b9826c337'.
  (fix conflicts and then run "git rebase --continue")
  (use "git rebase --skip" to skip this patch)
  (use "git rebase --abort" to check out the original branch)

Unmerged paths:
  (use "git restore --staged <file>..." to unstage)
  (use "git add <file>..." to mark resolution)
	both modified:   ivy/functional/backends/numpy/creation.py

no changes added to commit (use "git add" and/or "git commit -a")

(base) linjx@ethans-macbook ivy % git add ivy/functional/backends/numpy/creation.py
(base) linjx@ethans-macbook ivy % git status                                       
interactive rebase in progress; onto b9826c337
Last commands done (56 commands done):
   pick 54148c40b Improving PR dashboard filtering
   pick b919f3998 Add 'out' to each function of creation in numpy backend
  (see more in file .git/rebase-merge/done)
Next commands to do (111 remaining commands):
   pick c6e8d4dd2 Add out arg to the creation submodule of jax, tf, torch backends
   pick fbab973fe Add whitespaces around equals; delete out arg of meshgrid
  (use "git rebase --edit-todo" to view and edit)
You are currently rebasing branch 'master' on 'b9826c337'.
  (all conflicts fixed: run "git rebase --continue")

Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
	modified:   ivy/functional/backends/numpy/creation.py

(base) linjx@ethans-macbook ivy % git push
fatal: You are not currently on a branch.
To push the history leading to the current (detached HEAD)
state now, use

    git push origin HEAD:<name-of-remote-branch>

(base) linjx@ethans-macbook ivy % git push origin master
To https://github.com/egg-west/ivy.git
 ! [rejected]            master -> master (non-fast-forward)
error: failed to push some refs to 'https://github.com/egg-west/ivy.git'
hint: Updates were rejected because a pushed branch tip is behind its remote
hint: counterpart. Check out this branch and integrate the remote changes
hint: (e.g. 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
(base) linjx@ethans-macbook ivy % git pull
error: cannot pull with rebase: Your index contains uncommitted changes.
error: please commit or stash them.

RitujaPawas commented:
@egg-west Changes look good! I will take a final look and let you know!

RitujaPawas reviewed (approved):
Hey @egg-west , lgtm! thankyou for your contribution.
all the best!
